gensim/models/word2vec.py

```diff
@@ -1698,10 +1707,58 @@ def reset_weights(self, hs, negative, wv):
         """Reset all projection weights to an initial (untrained) state, but keep the existing vocabulary."""
         logger.info("resetting layer weights")
         wv.vectors = empty((len(wv.vocab), wv.vector_size), dtype=REAL)
+        
+        #if pre-trained embedding file is given, load it
+        p_emb = {}
+        syn0_oov = []
+        t = time.time()
+        if self.pretrained_emb == None:
+            logger.info("pretrained_emb is None!")
+
+        if self.pretrained_emb != None:
+            logger.info("loading pre-trained embeddings")
+            with smart_open.open(self.pretrained_emb) as fin:
```

Comment:
Way too nested here.

Move the loading code out to a separate function. Ideally, break it up into smaller functions.

Reply:
accepted!